core/field_dropdown.ts

```diff
@@ -114,30 +113,29 @@ export class FieldDropdown extends Field {
    * @throws {TypeError} If `menuGenerator` options are incorrectly structured.
    */
   constructor(
-      menuGenerator: AnyDuringMigration[][]|Function|Sentinel,
-      opt_validator?: Function, opt_config?: FieldConfig) {
+      menuGenerator: MenuGenerator, opt_validator?: Function,
+      opt_config?: FieldConfig);
+  constructor(menuGenerator: Sentinel);
+  constructor(
+      menuGenerator: MenuGenerator|Sentinel, opt_validator?: Function,
+      opt_config?: FieldConfig) {
     super(Field.SKIP_SETUP);
 
-    // If we pass SKIP_SETUP, don't do *anything* with the menu generator.
-    if (menuGenerator === Field.SKIP_SETUP) {
+    if (isMenuGenerator(menuGenerator)) {
+      if (Array.isArray(menuGenerator)) {
+        validateOptions(menuGenerator);
+        const trimmed = trimOptions(menuGenerator);
+        this.menuGenerator_ = trimmed.options;
+        this.prefixField = trimmed.prefix || null;
```

Comment:
Originally, I tried keeping the resulting values identical to the previous implementation in mind of a minor release. Since this will probably be included in a major release, though, I think it might be worth making `prefixField` and `suffixField` accept undefined instead of null. `null` is in a weird spot where it's mostly symbolic now, perhaps conveying an explicit choice to provide a non-value rather than choosing to not provide a value.

I'll make the update in favor of undefined since it keeps the code more concise, though I can adjust either way.

Reply:
Actually, updating `prefixField` and `suffixField` would require updating the `Field` parent class and impacted files so that'd be a pretty expansive change. Would it be fine to leave this as is for now? Ultimately I think those values probably should be just optional strings rather than string|null, though that's probably for a future PR.